modules/function/src/main/java/com/opengamma/strata/function/calculation/swaption/SwaptionCalculationMarketView.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.function.calculation.swaption;
+
+import com.opengamma.strata.calc.marketdata.CalculationMarketData;
+
+/**
+ * Market data for swaptions, used for calculation across multiple scenarios.
+ * <p>
+ * This interface exposes the market data necessary for pricing a swaption.
+ * This includes volatilities and rates.
+ * <p>
+ * Implementations of this interface must be immutable.
+ */
+public interface SwaptionCalculationMarketView {
+
+  /**
+   * Gets the lookup that provides access to swaption volatilities.
+   * 
+   * @return the swaption lookup
+   */
+  public abstract SwaptionMarketLookup getLookup();
+
+  /**
+   * Gets the market data.
+   * 
+   * @return the market data
+   */
+  public abstract CalculationMarketData getMarketData();
+
+  /**
+   * Returns a copy of this instance with the specified market data.
+   * 
+   * @param marketData  the market data to use
+   * @return a market view based on the specified data
+   */
+  public abstract SwaptionCalculationMarketView withMarketData(CalculationMarketData marketData);
```

Comment:
This allows the market data to be swapped out for another set, for example when bumping. This method matches the equivalent on rate market view, which is used in bucketed gamma.